lib/temporal.rb

```diff
@@ -81,6 +83,58 @@ def signal_workflow(workflow, signal, workflow_id, run_id, input = nil)
       )
     end
 
+    # Long polls for a workflow to be completed and returns whatever the execute function
+    # returned.
+    # run_id of nil: await the entire workflow completion.  This can span multiple runs
+    # in the case where the workflow uses continue as new.
+    def await_workflow_result(workflow, workflow_id:, run_id: nil, **args)
+      options = args.delete(:options) || {}
+      execution_options = ExecutionOptions.new(workflow, options)
+
+      current_run_id = run_id
+      loop do
+        history_response = client.get_workflow_execution_history(
+          namespace: execution_options.namespace,
+          workflow_id: workflow_id,
+          run_id: current_run_id,
+          wait_for_new_event: true,
+          event_type: :close
+        )
+        event = history_response['history']['events'].first
+        case event.event_type
+        when :EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED
+          payloads = event['workflow_execution_completed_event_attributes'].result
+          return nil if !payloads # happens when the workflow itself returns nil
+          return Temporal::Client::Converter::Payload::JSON.new.from_payload(payloads['payloads'].first)
```

Comment:
This is peeking into too low level API and also ignores any pre-set converters (such as encryption, etc). We need to use something higher level. @robholland can you please advice?

Reply:
The payload concern (https://github.com/coinbase/temporal-ruby/blob/75a3f854557ed47cbbdd4d115e40903b28b55d3b/lib/temporal/concerns/payloads.rb) should be used here: `from_result_payloads(payloads)`.